src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,183 @@
+package duke;
+
+import duke.exceptions.*;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.tools.Parser;
+import duke.tools.UI;
+
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Task manager with private attribute task array to store tasks.
+ * Public methods to read/write tasks
+ */
+public class TaskManager {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private final static UI ECHO_BACK = new UI();
+    private final static String MARKED_CAPTION = "      Nice! I've marked this task as done:";
+    private final static String NEW_TASK_CAPTION = "      Got it. I've added this task:";
+    private final static String DELETE_TASK_CAPTION = "      Noted! I've removed this task:c";
+    private final static String UNMARKED_CAPTION = "      OK, I've marked this task as not done yet:";
+    private final static String DATA_PATH = "duke_data.text";
+    private final static File DATA_FILE = new File(DATA_PATH);
+
+
+
+
+    public static void loadTask(Task task){
+        tasks.add(task);
+    }
+
+    public static ArrayList<Task> getALLTasks(){
+        return tasks;
+    }
+    /**
+     * Create new todoTask.
+     *
+     * @param taskDescription
+     * @return
+     */
+    public Todo createNewTodo(String taskDescription) throws IOException{
+        Todo newTodo = new Todo(taskDescription);
+        return newTodo;
+    }
+
+    /**
+     * Create new deadline task.
+     * Get deadline from command.
+     *
+     * @param taskDescription
+     * @return
+     */
+    public Deadline createNewDeadline(String taskDescription) throws MissingParameterException, IOException {
+        int index = taskDescription.indexOf("/by");
+        if(index==-1){
+            throw new MissingParameterException();
+        }
+        String deadlineContent = taskDescription.substring(0,index-1);
+        String deadlineDate = taskDescription.substring(index+4);
```

Comment:
Try to avoid magic literals. There has been a few more instances in the rest of your code. You can use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.